google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -302,6 +320,99 @@ String getServiceNameFromAppEngineWebXml(XmlFile appengineWebXml) {
+      PsiElement element = generateFromTemplate(
+          type == AppEngineFlexibleDeploymentArtifactType.JAR
+              ? AppEngineTemplateGroupDescriptorFactory.DOCKERFILE_JAR_TEMPLATE
+              : AppEngineTemplateGroupDescriptorFactory.DOCKERFILE_WAR_TEMPLATE,
+          "docker",
+          "Dockerfile",
+          FileTemplateManager.getDefaultInstance().getDefaultProperties(),
+          module);
+
+      if (element != null) {
+        // Remove the .docker extension to satisfy the Docker convention. This extension was added
+        // since the templating mechanism requires an extension or else a default template type of
+        // "java" will be assumed.
+        RenamePsiElementProcessor.DEFAULT.renameElement(
+            element,
+            "Dockerfile" /*newName*/,
+            UsageInfo.EMPTY_ARRAY,
+            null /*listener*/);
+      }
+    });
+  }
+
+  @Nullable
+  private PsiElement generateFromTemplate(String templateName, String templateDirectoryName,
+      String fileName, Properties templateProperties, Module module) {
+    FileTemplate configTemplate = FileTemplateManager.getDefaultInstance().getInternalTemplate(
+        templateName);
+    VirtualFile virtualFile = ModuleRootManager.getInstance(module).getContentRoots()[0]
+        .findFileByRelativePath("src/main");
+
+    if (virtualFile != null) {
+      PsiDirectory directory = PsiManager.getInstance(module.getProject())
+          .findDirectory(virtualFile);
+
+      if (directory != null) {
+        PsiDirectory configDirectory;
+        try {
+          directory.checkCreateSubdirectory(templateDirectoryName);
+          configDirectory = directory.createSubdirectory(templateDirectoryName);
+        } catch (IncorrectOperationException ioe) {
+          // checkCreateSubdirectory threw an exception suggesting that the directory may already
+          // exist. Skip creating the directory and attempt to write file.
+          configDirectory = directory.findSubdirectory(templateDirectoryName);
+        }
+
+        if (configDirectory != null
+            && FileTemplateUtil.canCreateFromTemplate(
+            new PsiDirectory[]{configDirectory}, configTemplate)) {
+          try {
+            return FileTemplateUtil.createFromTemplate(
+                configTemplate,
+                fileName,
+                templateProperties,
+                configDirectory);
+          } catch (Exception e) {
+            // If the file already exists, this exception will be thrown by createFromTemplate
+            // We want to silently skip the generation in this case.
+            logger.debug("Failed to create app.yaml from template. " + e.getMessage());
+          }
+        } else {
+          logger.error("Failed to create app.yaml from template in directory: " + configDirectory);
+        }
+      }
+    } else {
+      logger.error("Failed to located source root for app.yaml creation.");
```

Comment:
this triggers the whole error reporting flow right?  We don't want to do that just because the user doesn't have a src/main.